### ISSUE TITLE
Initialisation and custom location for HsFFI.h

### ISSUE THREAD

**Issue by mchakravarty** [mchakravarty/language-c-inline#46]
Cherry pick 7566546b245062128292e29b1705b74cc68e8db3 and 57eafb13032bee6fc74712b6d08ad9df5ea91b5a from the release/0.7 branch.

**Comment by cartazio** [comment]
has this been done yet?

**Comment by mchakravarty** [comment]
No. TBH I am still using the version on that branch and, in the meantime, I created more patches that need to be merged into master from that branch. I just haven't found the time for it 😞

**Comment by cartazio** [comment]
What about the converse and adding ghc 8 support to the older branch ?

**Comment by mchakravarty** [comment]
Should be possible, I assume. I didn't try using GHC 8 yet. What's going wrong?

**Comment by cartazio** [comment]
I didn't dig too deep but mostly some th constructors have an extra arg.
Your 0.8 branch seems to build fine though I've yet to test it.  Will dig
deeper tomorrow

**Comment by mchakravarty** [comment]
Do you mean the `release/0.7` branch? Let me know what you find.
